source/common/stats/thread_local_store.h

```diff
@@ -108,19 +110,22 @@ class ParentHistogramImpl : public MetricImpl<ParentHistogram> {
   const std::string bucketSummary() const override;
 
   // Stats::Metric
-  SymbolTable& symbolTable() override { return parent_.symbolTable(); }
+  SymbolTable& symbolTable() override;
   bool used() const override;
 
   // RefcountInterface
-  void incRefCount() override { refcount_helper_.incRefCount(); }
-  bool decRefCount() override { return refcount_helper_.decRefCount(); }
-  uint32_t use_count() const override { return refcount_helper_.use_count(); }
+  void incRefCount() override { ++ref_count_; }
+  bool decRefCount() override;
+  uint32_t use_count() const override { return ref_count_; }
+
+  // Indicates that the ThreadLocalStore is shutting down, so no need to clear its histogram_set_.
+  void setShuttingDown(bool shutting_down) { shutting_down_ = shutting_down; }
 
 private:
   bool usedLockHeld() const ABSL_EXCLUSIVE_LOCKS_REQUIRED(merge_lock_);
 
   Histogram::Unit unit_;
-  Store& parent_;
+  ThreadLocalStoreImpl& parent_;
```

Comment:
A bit of a re-think is needed about the `tls_scope_` entry here, as histograms can outlive the scopes they were created in, and new scopes can be added.

We really should be sharing the tls cache entries across all scopes that share a parent histogram, but it requires some diagramming to sort out the data structures.

There's no unit test that shows this issue yet, but it causes ads_integration_test to fail, thank god!